d3a_api_client/rest_market.py

```diff
@@ -47,14 +49,20 @@ def _on_market_cycle(self, message):
         logging.debug(f"A new market was created. Market information: {message}")
 
         def executor_function():
-            self.on_market_cycle(message)
+            try:
+                self.on_market_cycle(message)
+            except Exception as e:
+                logging.error(f"on_market_cycle raised exception: {e}. Traceback: {traceback.format_exc()}")
         self.callback_thread.submit(executor_function)
 
     def _on_finish(self, message):
         logging.debug(f"Simulation finished. Information: {message}")
 
         def executor_function():
-            self.on_finish(message)
+            try:
+                self.on_finish(message)
+            except Exception as e:
+                logging.error(f"on_finish raised exception: {e}. Traceback: {traceback.format_exc()}")
```

Comment:
Would be better IMO to have traceback starting from new line

Reply:
I agree, will add a newline.